util/data_generator.py

```diff
@@ -62,21 +67,21 @@ def get_random(self):
         return pt, ct, key
 
     def get_kmac_fixed(self):
-        pt = np.asarray(self.text_fixed)
-        key = np.asarray(self.key_fixed)
+        pt = self.text_fixed
+        key = self.key_fixed
         mac_fixed = KMAC128.new(key=bytes(self.key_fixed), mac_len=32)
         mac_fixed.update(bytes(self.text_fixed))
-        ct = np.asarray(bytearray(mac_fixed.digest()))
+        ct = mac_fixed.digest()
```

Comment:
I'm not a python expert and also got confused with the different byte representations, but I'm not sure if the byte/int/array problematic is fully covered here.

Imho `.update()` method requires byte strings or byte arrays and also returns a byte array. Are they dynamically casted?

Further, we later only use the ct to compare it in with the received one. AFAIK this also is done in bytearrays. So we should be fine here. Anyways, I guess it would be good to document where we use which kind of data representation. I'd suggest to put this information at the top of each script.